test-d/macros.ts

```diff
@@ -1,5 +1,6 @@
 import {expectType} from 'tsd';
-import test, {ExecutionContext, Macro} from '..'; // eslint-disable-line import/no-unresolved
+
+import test, {ExecutionContext, Macro} from '..';
```

Comment:
```suggestion
import test, {ExecutionContext, Macro} from '../index.js';
```

You should use full name and extension for the TS stuff too.

Reply:
There's is no `index.js` file, but there is an exports map. Which the linter rules don't seem to deal with, so I have direct imports of `entrypoints/main.mjs` elsewhere. But then TypeScript can't find the type definition. Hence `..` here.